Verbose logs for popErrorLevel

I am not quite sure why these two function are pushing out logs at warn/error levels. It just feeds the logs with repetitive useless messages in prod.

https://github.com/salesagility/SuiteCRM/blob/master/include/StateSaver.php#L302-L329

Just noting here that (I believe) this issue is about the non-stop outputting of

```
Wed Mar 13 12:13:46 2019 [1642][1][WARN] Saving error level. Try to remove the error_reporting() function from your code.
Wed Mar 13 12:13:46 2019 [1642][1][ERROR] Pop error level. Try to remove the error_reporting() function from your code.
```

It looks like there are couple of places that modules (recently) have set `error_reporting()` which shouldn't be.

--- a/README.md
+++ b/README.md
@@ -1,8 +1,8 @@
 ## SuiteCRM 7.10.14
 
-[![Build Status](https://travis-ci.org/salesagility/SuiteCRM.svg?branch=hotfix)](https://travis-ci.org/salesagility/SuiteCRM)
+[![Build Status](https://travis-ci.org/salesagility/SuiteCRM.svg?branch=hotfix-7.10.x)](https://travis-ci.org/salesagility/SuiteCRM)
 [![codecov](https://codecov.io/gh/salesagility/SuiteCRM/branch/hotfix/graph/badge.svg)](https://codecov.io/gh/salesagility/SuiteCRM/branch/hotfix)
-
+[![Gitter chat](https://badges.gitter.im/gitterHQ/gitter.png)](https://gitter.im/suitecrm/Lobby)	
 
 ### What's in this repository ###
 

--- a/include/StateChecker.php
+++ b/include/StateChecker.php
@@ -225,9 +225,7 @@ class StateChecker
         $this->lastHash = $hash;
 
         if (!$this->checkHash($hash, $key)) {
-            if ($key != 'errlevel') { // TODO: temporary remove the error level check from state
-                throw new StateCheckerException('Hash doesn\'t match at key "' . $key . '".');
-            }
+            throw new StateCheckerException('Hash doesn\'t match at key "' . $key . '".');
         }
         
         if (StateCheckerConfig::get('saveTraces')) {

--- a/include/StateSaver.php
+++ b/include/StateSaver.php
@@ -291,31 +291,38 @@ class StateSaver
             }
         }
     }
-    
+
     /**
      * Save Error Reporting Level into the store at an optional key and namespace.
      * (note: error level should not be changed for any reason, so use it for own risk)
      *
      * @param string $key
      * @param string $namespace
+     * @param bool $doLogging
+     * @throws StateSaverException
      */
-    public function pushErrorLevel($key = 'level', $namespace = 'error_reporting')
+    public function pushErrorLevel($key = 'level', $namespace = 'error_reporting', $doLogging = true)
     {
-        LoggerManager::getLogger()->warn('Saving error level. Try to remove the error_reporting() function from your code.');
+        if ($doLogging) {
+            LoggerManager::getLogger()->warn('Saving error level. Try to remove the error_reporting() function from your code.');
+        }
         $level = error_reporting();
         $this->push($level, $key, $namespace);
     }
-    
+
     /**
      * Restore Error Reporting Level from the store at an optional key and namespace.
      * (note: error level should not be changed for any reason, so use it for own risk)
      *
      * @param string $key
      * @param string $namespace
+     * @param bool $doLogging
      */
-    public function popErrorLevel($key = 'level', $namespace = 'error_reporting')
+    public function popErrorLevel($key = 'level', $namespace = 'error_reporting', $doLogging = true)
     {
-        LoggerManager::getLogger()->error('Pop error level. Try to remove the error_reporting() function from your code.');
+        if ($doLogging) {
+            LoggerManager::getLogger()->error('Pop error level. Try to remove the error_reporting() function from your code.');
+        }
         $level = $this->pop($key, $namespace);
         error_reporting($level);
     }

--- a/include/Sugar_Smarty.php
+++ b/include/Sugar_Smarty.php
@@ -157,17 +157,16 @@ class Sugar_Smarty extends Smarty
         $this->assign('MOD', $mod_strings);
         $this->assign('APP_CONFIG', $sugar_config);
 
+        $state = new SuiteCRM\StateSaver();
+        $state->pushErrorLevel('sugar_smarty_errors', 'error_reporting', false);
+
         if (!(isset($sugar_config['developerMode']) && $sugar_config['developerMode'])) {
             $level = isset($sugar_config['smarty_error_level']) ? $sugar_config['smarty_error_level'] : 0;
-            $errorLevelStored = error_reporting();
             error_reporting($level);
         }
         $fetch = parent::fetch(get_custom_file_if_exists($resource_name), $cache_id, $compile_id, $display);
-        if (!(isset($sugar_config['developerMode']) && $sugar_config['developerMode'])) {
-            $level = isset($sugar_config['smarty_error_level']) ? $sugar_config['smarty_error_level'] : 0;
-            error_reporting($errorLevelStored);
-        }
-        
+        $state->popErrorLevel('sugar_smarty_errors', 'error_reporting', false);
+
         return $fetch;
     }
 

